### PR TITLE
Bump webdav to ^4

### PR DESCRIPTION
**Pull request by skjnldsv**
Removed unused service since 3 years...
https://github.com/nextcloud/photos/pull/4/files#diff-239342c3890dc7339f0c87a00179a78c0502d3f74b5b387f18fd2e00cb19d194L57

Webdav 3 is deprecated and don't receive updates anymore

### CODE REVIEW

**Comment by marcelklehr** [comment]
Does the client API stay the same? That's nice!

**Comment by skjnldsv** [comment]
> Does the client API stay the same? That's nice!

Yes and no, I had to adapt a few hacks we do as well as their new esm export.

File: webpack.js [hunk 1]
```diff
@@ -41,9 +42,10 @@ webpackConfig.module.rules = Object.values(webpackRules)
 webpackConfig.plugins.push(
 	// patch webdav/dist/request.js
 	new webpack.NormalModuleReplacementPlugin(
-		/request.js/,
+		/request(\.js)?/,
```
Comment (marcelklehr) [inline webpack.js:45]: oof, I didn't know about this. :'D

**Review by marcelklehr — APPROVED** [review]

File: webpack.js (hunk 1 above)
Comment (skjnldsv) [reply webpack.js:45]: We could make it upstream, but https://github.com/perry-mitchell/webdav-client/issues/179

File: webpack.js [hunk 2]
```diff
@@ -41,9 +42,10 @@ webpackConfig.module.rules = Object.values(webpackRules)
...
 		function(resource) {
-			if (resource.context.indexOf('webdav/dist') > -1) {
+			if (resource.context.indexOf('webdav') > -1) {
+				console.debug('Patched request for webdav', basename(resource.contextInfo.issuer))
```
Comment (marcelklehr) [inline webpack.js:48]: This doesn't seem to work on latest master, though :S I've put a direct throw Error into patchedRequest and it never gets thrown?

**Pull request merged**